src/sentry/users/models/user_merge_verification_code.py

```diff
@@ -0,0 +1,57 @@
+import secrets
+from datetime import datetime, timedelta
+
+from django.conf import settings
+from django.db import models
+from django.utils import timezone
+
+from sentry.backup.scopes import RelocationScope
+from sentry.db.models import FlexibleForeignKey, Model, control_silo_model, sane_repr
+from sentry.users.models.user import User
+from sentry.users.services.user.model import RpcUser
+
+TOKEN_MINUTES_VALID = 30
+
+
+def generate_token() -> str:
+    return secrets.token_hex(nbytes=32)
+
+
+def generate_expires_at() -> datetime:
+    return timezone.now() + timedelta(minutes=TOKEN_MINUTES_VALID)
+
+
+@control_silo_model
+class UserMergeVerificationCode(Model):
```

Comment:
It's generally helpful to use `DefaultFieldsModel` here so that you get date_added/updated.